pkg/build/api/types.go

```diff
@@ -401,6 +401,9 @@ type GitBuildSource struct {
 
 	// HTTPSProxy is a proxy used to reach the git repository over https
 	HTTPSProxy *string
+
+	// NoProxy is the list of domains for which the proxy should not be used
+	NoProxy *string
```

Comment:
if it's a var to the build then:
1) we have to whitelist it so it can be set on the privileged container
2) all operations on the privileged container will use that value which makes it riskier(?) and also less granular control over proxy behavior
3) build env vars get used during assemble which might be undesirable
4) build env vars get baked into the final application image which might be undesirable

Also the ship has kinda already sailed w/ the httpproxy/httpsproxy fields so it makes sense to me to put the noproxy field in the same place.